api/topic/topic.go

```diff
@@ -401,16 +431,21 @@ func Insert(topic *tat.Topic, user *tat.User) error {
 	store.EnsureIndexesMessages(topic.Collection)
 
 	h := fmt.Sprintf("create a new topic :%s", topic.Topic)
-	err = addToHistory(topic, bson.M{"_id": topic.ID}, user.Username, h)
+	err = addToHistory(topic, bson.M{"_id": topic.ID}, u.Username, h)
 	if err != nil {
 		log.Errorf("Error while inserting history for new topic %s", err)
 	}
 
-	return AddRwUser(topic, user.Username, user.Username, false)
+	return AddRwUser(topic, u.Username, u.Username, false)
 }
 
 // Delete deletes a topic from database
-func Delete(topic *tat.Topic, user *tat.User) error {
+func Delete(topic *tat.Topic, u *tat.User) error {
```

Comment:
user is useless here. Same than Insert, we have to clean all cache about topics, for all users.